### DISCUSSION TITLE
How to increase and decrease product stock quantity while purchasing and selling products?

### DISCUSSION THREAD

**Discussion by imhemayatsangin** [serenity-is/Serenity#6311] [Help]
Greetings folks,
Can anybody help me out how to handle the product stock quantity transaction, for example:
I have particular products and I have 2 forms one is Stock in(Purchase) and another form is stock out(Sell).
At first the quantity for a particular product is 0 and while stock in we purchase that product of  quantity 100, so in the second form stock out when we stock out or sell that particular product of 50 items so the available stock should decrease and it should only show available quantity 50. 
How to do it and please if anybody has done same thing then please share the code with me.
Looking forward to hearing from you guys please!

**Comment by imhemayatsangin** [comment]
Dear @volkanceylan @dfaruque @VictorTomaili , Please help me in this issue, I will be thankful to you.

**Reply by reach2rv** [reply]
Hi @sangincode404  this can be done by overriding AfterSave of Purchase and Sales repository where you assign data to Stock entity with qty value as add or deduct based on purchase sale transaction. and then call create method of stock repository with entity created.

Hope this help.
